cupy/core/core.pyx

```diff
@@ -2363,6 +2378,14 @@ cpdef ndarray _convert_object_with_cuda_array_interface(a):
         dev_id = device.get_device_id()
     mem = memory_module.UnownedMemory(ptr, nbytes, a, dev_id)
     memptr = memory.MemoryPointer(mem, 0)
+    # the v3 protocol requires an immediate synchronization, unless
+    # 1. the stream is not set (ex: from v0 ~ v2) or is None
+    # 2. users explicitly overwrite this requirement
+    stream_ptr = desc.get('stream')
+    if stream_ptr is not None:
+        if _util.CUDA_ARRAY_INTERFACE_SYNC:
+            stream = cuda.stream.ExternalStream(stream_ptr)
+            stream.synchronize()
```

Comment:
This is done mainly to emphasize that `stream_ptr` is likely coming from external libraries. If desired I can simplify it to `runtime.streamSynchronize(stream_ptr)`, which should still allow mock tests I think.

Reply:
Yes, it can be simplified to `runtime.streamSynchronize(stream_ptr)`, I think!

Reply:
Hi @asi1024 Looks like I can't mock `cpdef` functions...😢 Should I leave this as is?

Reply:
One workaround I know is defining `def` function just calling `runtime.streamSynchronize` here, but I'm wondering if it causes performance degradation. I think it's OK to leave this as is without mock tests.

Reply:
Sounds good, thanks.